cpp/arcticdb/version/symbol_list.cpp

```diff
@@ -243,10 +232,25 @@ bool SymbolList::can_update_symbol_list(const std::shared_ptr<Store>& store,
         }
     }
 
-    SymbolList::KeyVector SymbolList::get_all_symbol_list_keys(const std::shared_ptr<StreamSource>& store) const {
+    SymbolList::KeyVector SymbolList::get_all_symbol_list_keys(const std::shared_ptr<StreamSource>& store) {
         std::vector<AtomKey> output;
+        uint64_t uncompacted_keys_found = 0;
         store->iterate_type(KeyType::SYMBOL_LIST, [&] (auto&& key) -> void {
-            output.push_back(to_atom(key));
+            auto atom_key = to_atom(key);
+            if(atom_key.id() != compaction_id) {
+                uncompacted_keys_found++;
+            }
+            if (uncompacted_keys_found == this->max_delta_ * 2 && !warned_expected_slowdown_) {
+                log::version().warn(
+                        "Warning - symbol list cache is significantly out of date. "
```

Comment:
The `Warning -` bit seems redundant.

Also, "significantly out of date" may lead to confusion. If the library has very frequent writes, even if the cache has been compacted seconds before, this message can still be triggered.